### DISCUSSION TITLE
Use dotnet nuget instead of nuget.exe and mono for binary caching

### DISCUSSION THREAD

**Discussion by sweemer** [microsoft/vcpkg#40692] [General]
Migrating this from https://github.com/microsoft/vcpkg/discussions/22684

I saw the following today at www.mono-project.com:

> we recommend that active Mono users and maintainers of Mono-based app frameworks migrate to .NET which includes work from this fork.

This suggests to me that it is finally time to switch the binary caching implementation to use dotnet nuget instead of nuget.exe and mono.

**Comment by jimwang118** [comment]
Duplicate of https://github.com/microsoft/vcpkg/discussions/22684.

**Comment by sweemer** [comment]
Hi @jimwang118 I converted the original discussion into an issue because it seems like there is clearer guidance on the migration now than there was two years ago. If you think that more discussion is required then can you please help to loop in the right people?

**Comment by dpogue** [comment]
This is especially relevant now in light of the GitHub Actions Ubuntu 24.04 runners [dropping Mono and NuGet as part of the standard image](https://github.com/actions/runner-images/issues/11138), Mono being deprecated, and the NuGet folks [encouraging everyone to switch to `dotnet nuget`](https://github.com/NuGet/setup-nuget/issues/168#issuecomment-2573539599)

**Comment by DimRochette** [comment]
Binary caching is mandatory in our system and we rely on vcpkg nuget to do that. 
It's our current single blocking point on our migration of build to ubuntu 26.04, It's currently funny to say to the team we won't migrate to newer gcc because of mono.
Even if we succeeded installing the new mono we still have issues on all httpS connections and we can't drop it.
it will be a really nice improvement for us to move to donet nuget.
